### PR TITLE
Support "submitOnChange" in the FileTree widget

### PR DESCRIPTION
**Pull request by AlexejKossmann**
This PR adds submitOnChange to FileTree Widget after selecting a file from the filesystem.

Recreated for #1436

### CODE REVIEW

**Comment by ausi** [comment]
I think it would be better to add ` onchange="' . $this->onchange . '"` to https://github.com/contao/contao/blob/e2ffafc889cfbfd2a227822faed7d00840196e87/core-bundle/src/Resources/contao/widgets/FileTree.php#L337

@leofeyer any reason why this was considered a feature in #1436 and not a bugfix?

**Comment by leofeyer** [comment]
I was not sure about unintended side-effects, therefore I tagged it as feature. I am fine with fixing it in Contao 4.9 as well.

**Review by aschempp — DISMISSED** [review]

**Review by leofeyer — CHANGES_REQUESTED** [review]
@AlexejKossmann Can you adjust the PR according to @ausi's [suggestions](https://github.com/contao/contao/pull/2878#issuecomment-800064971) please?

**Comment by AlexejKossmann** [comment]
@ausi @leofeyer i have tested the onchange attribute in 4.9, it did not submit on selecting a new file. As i considder this is not supported on input `type="hidden"`  ?

File: core-bundle/src/Resources/contao/widgets/FileTree.php [hunk 1]
```diff
@@ -371,7 +371,7 @@ public function generate()
               onSuccess: function(txt, json) {
                 $("ctrl_' . $this->strId . '").getParent("div").set("html", json.content);
                 json.javascript && Browser.exec(json.javascript);
-                $("ctrl_' . $this->strId . '").fireEvent("change");
+                $("ctrl_' . $this->strId . '").fireEvent("change");' . ($this->arrConfiguration['submitOnChange'] ? ' Backend.autoSubmit("' . $this->strTable . '");' : '') . '
```
Comment (ausi) [inline core-bundle/src/Resources/contao/widgets/FileTree.php]:
```suggestion
                var evt = document.createEvent('HTMLEvents');
                evt.initEvent('change', true, true);
                $("ctrl_' . $this->strId . '").dispatchEvent(evt);
```

File: core-bundle/src/Resources/contao/widgets/FileTree.php [hunk 2]
```diff
@@ -334,7 +334,7 @@ public function generate()
 		$strSet = implode(',', array_map('StringUtil::binToUuid', $arrSet));
 		$strOrder = $blnHasOrder ? implode(',', array_map('StringUtil::binToUuid', $this->{$this->orderField})) : '';
 
-		$return = '<input type="hidden" name="' . $this->strName . '" id="ctrl_' . $this->strId . '" value="' . $strSet . '">' . ($blnHasOrder ? '
+		$return = '<input type="hidden" name="' . $this->strName . '" id="ctrl_' . $this->strId . '" value="' . $strSet . '" onchange="' . $this->onchange . '">' . ($blnHasOrder ? '
```
Comment (ausi) [inline core-bundle/src/Resources/contao/widgets/FileTree.php]:
```suggestion
		$return = '<input type="hidden" name="' . $this->strName . '" id="ctrl_' . $this->strId . '" value="' . $strSet . '"' . ($this->onchange ? ' onchange="' . $this->onchange . '"' : '') . '>' . ($blnHasOrder ? '
```

**Review by ausi — COMMENTED** [review]
> i have tested the onchange attribute in 4.9, it did not submit on selecting a new file.

Could you please test if the changes from my comments work in your case?

File: core-bundle/src/Resources/contao/widgets/FileTree.php (hunk 2 above)
Comment (AlexejKossmann) [reply core-bundle/src/Resources/contao/widgets/FileTree.php]: @ausi yes this helped, your changes are working

**Review by ausi — DISMISSED** [review]

**Comment by ausi** [comment]
@AlexejKossmann Can you please rebase this pull request on the 4.9 branch and switch the target branch to 4.9?

**Comment by AlexejKossmann** [comment]
@ausi so i did the rebase, was this correct?

**Comment by ausi** [comment]
The result doesn’t look right, but I can try to fix it if you like?

**Comment by AlexejKossmann** [comment]
Yes this would be nice, seems like the rebase added me to every comit i rebased

**Comment by ausi** [comment]
Done.

**Comment by AlexejKossmann** [comment]
Thanks alot, what did i wrong on the rebase? Was my first rebase on a such big codebase

**Comment by ausi** [comment]
> what did i wrong on the rebase?

Hard to say. It looks like you rebased the 4.x branch (instead of your changes only) onto 4.9. And you added several unrelated files in ac67a3e59c5b1676e716a6911660179b341b17d5 (.idea folder and many language files). And in the last commit d9b2cc7a7a956d7f05f4c556f58e5b886b380df4 you merged the 4.x branch.

**Review by ausi — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @AlexejKossmann.

**Pull request merged**